src/components/TimeSlicing.vue

```diff
@@ -0,0 +1,263 @@
+        allVars.delete('source');
+        allVars.delete('target');
+        allVars.delete('index');
+
+        return cleanVariableList(allVars);
+      }
+      return new Set();
+    });
+
+    // Compute the min and max times
+    const timeRange = computed(() => {
+      const range: number[] = [0, 0];
+      if (startTimeVar.value !== null && endTimeVar.value !== null && originalNetwork.value !== null) {
+        // Loop through all edges, return min and max time values
+        originalNetwork.value.edges.forEach((edge: Edge) => {
+          if (edge[startTimeVar.value] < range[0]) {
+            range[0] = edge[startTimeVar.value];
+          }
+          if (edge[endTimeVar.value] > range[1]) {
+            range[1] = edge[endTimeVar.value];
+          }
+        });
+      }
+      return range;
+    });
+
+    const timeMin = computed(() => timeRange.value[0]);
+    const timeMax = computed(() => timeRange.value[1]);
+
+    const selectedRange = ref([0, 0]);
+
+    watch([timeMax], () => {
+      if (timeMax.value > 0) { selectedRange.value = [timeMin.value, timeMax.value]; }
+    });
+
+    function sliceNetwork() {
+      let networkToReturn: SlicedNetworks[] = [];
+
+      // Resets to original network view
+      if (originalNetwork.value !== null) {
+        if (timeSliceNumber.value === 1) {
+          store.commit.setSlicedNetwork(networkToReturn);
+          store.commit.setNetwork(originalNetwork.value);
+          return networkToReturn;
+        }
+        if (originalNetwork.value !== null) {
+          const slicedNetwork: SlicedNetworks[] = [];
+          const timeInterval = (selectedRange.value[1] - selectedRange.value[0]) / timeSliceNumber.value;
+
+          // Generate time chunks
+          // eslint-disable-next-line no-plusplus
+          for (let i = 0; i < timeSliceNumber.value; i++) {
+            const currentSlice: SlicedNetworks = { slice: i, time: [0, 0], network: { nodes: [], edges: [] } };
+            currentSlice.time = [i * timeInterval, (i + 1) * timeInterval];
+            currentSlice.network.nodes = originalNetwork.value.nodes;
+            slicedNetwork.push(currentSlice);
+          }
+
+          // Generate sliced network
+          let i = 0;
+          originalNetwork.value.edges.forEach((edge: Edge) => {
+            if (edge[startTimeVar.value] >= slicedNetwork[i].time[0] && edge[startTimeVar.value] < slicedNetwork[i].time[1]) {
+              slicedNetwork[i].network.edges.push(edge);
+            } else if (i < timeSliceNumber.value) {
+              i += 1;
+              slicedNetwork[i].network.edges.push(edge);
+            }
+          });
+          networkToReturn = slicedNetwork;
+
+          store.commit.setSlicedNetwork(networkToReturn);
+          store.commit.setNetwork(networkToReturn[0].network);
+          return networkToReturn;
+        }
+        store.commit.setSlicedNetwork(networkToReturn);
+        return networkToReturn;
+      }
+      return networkToReturn;
```

Comment:
There are a ton of different places where you can return from this function, which makes it tricky to read and follow. Could you try refactoring this so that there is just one return statement? You'd use more if statements in the body to manage the flow and then at the end, just return the `networkToReturn` after all the ifs

Reply:
Wow yeah that code was so unnecessarily complicated -- let me know what you think of the simplified version

Reply:
Looks much better 😄